conda/core/package_cache.py

```diff
@@ -432,7 +437,12 @@ def make_actions_for_dist(dist, record):
                 target_pkgs_dir=first_writable_cache.pkgs_dir,
                 target_extracted_dirname=dist.dist_name,
             )
-            return cache_axn, extract_axn
+            dr_axn = DumpRecordAction(
+                target_pkgs_dir=first_writable_cache.pkgs_dir,
+                target_extracted_dirname=dist.dist_name,
+                record=record,
+            )
+            return cache_axn, extract_axn, dr_axn
```

Comment:
Yeah, we shouldn't create that file more than once (I think that's what I had in mind with my earlier question this afternoon about the data flow and extraction and downloading being potentially separate transactions). So when the file already exists, the action should be a no-op.